scripts/Muon/GUI/ElementalAnalysis/PeriodicTable/periodic_table_model.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import print_function, absolute_import
+
+import os
+import json
+
+from Muon.GUI.Common import message_box
+from Muon.GUI import ElementalAnalysis
+
+
+class PeriodicTableModel(object):
+    def __init__(self):
+        self._peak_data = {}
+        self._peak_data_file = os.path.join(
+            os.path.dirname(
+                ElementalAnalysis.__file__),
+            "peak_data.json")
+        self.load_peak_data()
+
+    def load_peak_data(self):
+        # using os.path.isfile allows file modifications to take place before opening:
+        # we don't want this.
+        try:
+            with open(self._peak_data_file, "r") as f:
+                try:
+                    self._peak_data = json.load(f)
+                except ValueError as error:
+                    message_box.warning(error)
+        except IOError as error:
```

Comment:
this wont catch all possible errors. You will need a ```except Exception as error``` at the end to make sure you get everything

Reply:
So I guess I could combine this into one larger try; catch that encompasses both the with open() and the dict lookup?

Reply:
yep